contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -159,7 +164,12 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
         override
         public
     {
-        require(transactionContext.ovmNUMBER == 0, "Only callable at the start of a transaction");
+        // Make sure that run() is not re-enterable.  This condition should awlways be satisfied
+        // Once run has been called once, due to the behvaior of _isValidInput().
+        if (transactionContext.ovmNUMBER != DEFAULT_UINT256) {
+            return;
```

Comment:
Hmmm is it ok to return here? If so, why were we reverting before?

Reply:
I bellieve we explicitly should not have been reverting here, because it would've allowed a malicious caller to revert with malformed data.